src/Kunstmaan/AdminBundle/Helper/IsAdminRouteHelper.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Kunstmaan\AdminBundle\Helper;
+
+/**
+ * Class IsAdminRouteHelper
+ */
+class IsAdminRouteHelper
+{
+
+    /**
+     * @var string $adminkey
+     */
+    private $adminKey;
+
+    /**
+     * @param string $adminKey
+     */
+    public function __construct($adminKey)
+    {
+        $this->adminKey = $adminKey;
+    }
+
+    /**
+     * Checks wether the given url poins to an admin route
+     * @param string $url
+     *
+     * @return bool
+     */
+    public function isAdminRoute($url)
+    {
+        //If the url contains an admin part and a preview part then it is not an admin route
+        preg_match(sprintf('/^(\/app_[a-zA-Z]+\.php)?\/([a-zA-Z_-]{2,5}\/)?%s(\/.*)?\/preview/', $adminKey), $url, $matches);
+
+        if (count($matches) > 0) {
+            return false;
+        }
+
+        preg_match(sprintf('/^\/(app_[a-zA-Z]+\.php\/)?([a-zA-Z_-]{2,5}\/)?%s\/(.*)/', $adminKey), $url, $matches);
```

Comment:
`$adminKey` is unsafe in the regex context (for example, it could contain a forward slash), should be escaped. Now `foo/admin` would trigger an error. 

It would be nice too if the value were normalized, i.e. leading and trailing slashes would be handled. Should one provide `admin` or `/admin` or `admin/`…?